app.rb

```diff
@@ -58,30 +58,37 @@ def authorized?
       body = "timestamp=#{Time.now.iso8601}"
       resp = conn.post(config.attempts_api_path, body)
 
-      encrypted_data = Base64.strict_decode64(resp.body)
-      iv = Base64.strict_decode64(resp.headers['x-payload-iv'])
-      encrypted_key = Base64.strict_decode64(resp.headers['x-payload-key'])
-
-      private_key = OpenSSL::PKey::RSA.new(config.attempts_private_key)
-      key = private_key.private_decrypt(encrypted_key)
-
-      decrypted = decrypt_attempts_response(
-        encrypted_data: encrypted_data, key: key, iv: iv,
-      )
-
-      events = JSON.parse(decrypted)
-      events && events.each do |_jti, jwes|
-        jwes.each do |_key_id, jwe|
-          begin
-            decrypted_events << JSON.parse(JWE.decrypt(jwe, config.attempts_private_key))
-          rescue
-            puts 'Failed to parse/decrypt event!'
+      if resp.status == 200
+        encrypted_data = Base64.strict_decode64(resp.body)
+        iv = Base64.strict_decode64(resp.headers['x-payload-iv'])
+        encrypted_key = Base64.strict_decode64(resp.headers['x-payload-key'])
+
+        private_key = OpenSSL::PKey::RSA.new(config.attempts_private_key)
+        key = private_key.private_decrypt(encrypted_key)
+
+        decrypted = decrypt_attempts_response(
+          encrypted_data: encrypted_data, key: key, iv: iv,
```

Comment:
this seems like a good candidate to move into a separate class `AttemptsApiClient` or something, maybe in a separate PR if this one is urgent

Reply:
Yeah, sure I will move this to a Separate client in the next PR.